data/transactions/logic/eval_test.go

```diff
@@ -35,6 +35,44 @@ import (
 	"github.com/algorand/go-algorand/protocol"
 )
 
+func defaultEvalProto() config.ConsensusParams {
+	return config.ConsensusParams{LogicSigVersion: 1, LogicSigMaxCost: 20000}
+}
+
+func defaultEvalParams() EvalParams {
+	proto := defaultEvalProto()
+	return EvalParams{Proto: &proto, Trace: &strings.Builder{}, Txn: &transactions.SignedTxn{}}
```

Comment:
Trace should default to nil; OR defaultEvalParams could take a `sb *strings.Builder` passed in. But if that's not done, then everything that now calls `defaultEvalParams()` should drop its `sb strings.Builder{}` because that won't point to anything useful, but was used for error reporting, and we shouldn't confuse ourselves with error reporting that doesn't work.

Reply:
addressed.